File: front_end/src/service/ThuongHieuService.js
import axios from "axios";

const api = 'http://localhost:8080/thuong-hieu'

export const listThuongHieu = (pageable) => {
    return axios.get(api, {
        params: { page: pageable.page,size: pageable.size}
    });
};

export const createThuongHieu = (thuongHieu) => axios.post(api, thuongHieu);
export const getThuongHieu = (thuongHieuId) => axios.get(api + '/' + thuongHieuId);
export const updateThuongHieu = (thuongHieuId, thuongHieu) => axios.put(api + '/' + thuongHieuId, thuongHieu);
export const deleteThuongHieu = (thuongHieuId) => axios.delete(api + '/' + thuongHieuId);

File: src/main/java/com/example/shoe/service/ThuongHieuService.java
package com.example.shoe.service;

import com.example.shoe.dto.request.ThuongHieuRequest;
import com.example.shoe.dto.response.ThuongHieuResponse;
import com.example.shoe.entity.ThuongHieu;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
public interface ThuongHieuService {
    Page<ThuongHieu> getThuongHieu(Pageable pageable);
    ThuongHieu createThuongHieu(ThuongHieuRequest request);
    ThuongHieuResponse getThuongHieuId(Integer id);
    ThuongHieuResponse updateThuongHieu(Integer id, ThuongHieuRequest request);
    void deleteThuongHieu(Integer id);
    Page<ThuongHieu> searchThuongHieu(String keyword, Pageable pageable);
}
